### PR TITLE
Added template support for Prawn PDF

### PR DESCRIPTION
**Pull request by kbrock**
Adds #33 by @agios which uses a more recent prawn.

Sure, this version of prawn is still very old, and the gem does not look maintained, but this PR focuses on removing warnings that show up with recent ruby implementations.

This branch of the PR removes the conflict and updates the version of prawn and adds the table gem. But it is largely not my work. So thank you Agios.

The pdf test has been disabled because it seems to fail on every upgrade.
Will look into the issue behind this. The most recent failure is due to significant decimal places for 0 and some font tweaks.

The previous sample pdf had a table with headers but no data. Since this causes problems with later prawn versions, I added a data row.

### CODE REVIEW

**Comment by kbrock** [comment]
@jrafanie this is generating a bunch of warnings in ruby 2.6 (let alone 2.7)
So I wanted to merge this PR (and then run a sweep to clean up the remaining warnings and release)

Do you have any comments or concerns for this?
It probably won't concern us too much since this doesn't touch the models.

**Comment by kbrock** [comment]
> What's in our custom gem? I don't remember what feature we fixed or added.

- https://github.com/ManageIQ/ruport/pull/1 - here: https://github.com/ruport/ruport/pull/34 
- fastercsv - done
- fixing some tests for ruby 1.9 - done

**Comment by kbrock** [comment]
kicking

**Comment by kbrock** [comment]
The decrease in coverage is right.
A lot of code was added to prawn_pdf but tests were not added.
I wonder if there is a way to make the file test more stable.

probably best way to test this stuff is to add some mocking / stubs

File: test/prawn_pdf_formatter_test.rb [hunk 1]
```diff
@@ -41,10 +41,15 @@ def test_render_pdf_basic
     data.column_names = %w[a b]
 
     data.to_prawn_pdf
+  end
 
-    expected_output = IO.read(File.join(__dir__, 'expected_outputs/prawn_pdf_formatter/pdf_basic.pdf.test')).bytes
+  def xtest_render_pdf_basic
```
Comment (jrafanie) [inline test/prawn_pdf_formatter_test.rb:46]: I'd prefer a comment as to why it's disabled here.  Someone might assume this was a typo and try to fix it not knowing it's intentionally disabled.  Maybe use the `skip` method in minitest?  

https://www.paperlesspost.com/blog/teams/skipping-tests-minitest/

**Comment by kbrock** [comment]
@jrafanie 

1. I changed the pdf we generated to not have an empty table. Certain versions of prawn were not handling that well. To be honest, I don't think an empty data table is that valid of a use case anyway.
2. Every time we change the version of prawn, it changes the pdf output. E.g.: it changed the number of decimal places when specifying x and y.

**Comment by kbrock** [comment]
reenabled the test

**Pull request merged**